metrics/ci_integrations/lib/integration/validation/model/field_validation_result.dart

```diff
@@ -6,55 +6,77 @@ import 'package:ci_integration/integration/validation/model/field_validation_con
 import 'package:equatable/equatable.dart';
 
 /// A class that represents a validation result for a single [Config]'s field.
-class FieldValidationResult extends Equatable {
+class FieldValidationResult<T> extends Equatable {
   /// A [FieldValidationConclusion] of this field validation result.
   final FieldValidationConclusion conclusion;
 
   /// A [String] containing additional information about this field validation
   /// result.
   final String additionalContext;
 
+  /// A validated data of this field validation result.
+  final T data;
+
+  /// Indicates whether this field validation result is successful.
```

Comment:
```suggestion
  /// Indicates whether this is a successful validation result.
```
Would it be a bit more clear?